bin/docker_entrypoint.sh

```diff
@@ -1,7 +1,10 @@
-#!/bin/bash
+#!/usr/bin/bash
+set -e
 
 echo "Rebuilding hnsw to ensure architecture compatibility"
 pip install --force-reinstall --no-cache-dir chroma-hnswlib
+
 export IS_PERSISTENT=1
 export CHROMA_SERVER_NOFILE=65535
-exec uvicorn chromadb.app:app --workers 1 --host 0.0.0.0 --port 8000 --proxy-headers --log-config chromadb/log_config.yml --timeout-keep-alive 30
+echo "Starting server with args: ${@}"
+exec uvicorn chromadb.app:app ${@}
```

Comment:
@MrZoidberg, I don't think we should proceed with this change. The main reason for this is that not having sensible defaults (which by the way can be exposed via ENV vars), will result in a broken docker image where people will not be able to do: 

```bash
docker run -d --rm --name chromadb -p 8000:8000  chromadb/chroma:latest
```

Reply:
ENTRYPOINT is the process that's executed inside the container. CMD is the default set of arguments that are supplied to the ENTRYPOINT process.
By stripping executable out of CMD I think you're making container a bit more secure as it's harder to ovveride executable.
I made this change because in the current Chroma docker you CANNOT override the port easily.

With this change both commands works (build with `docker build -t server .`):
- `docker run -it --rm --name chromadb -p 8000:8000 server` --> results in chroma running on port 8000
- `docker run -it --rm --name chromadb -p 8002:8002 server --workers 1 --host 0.0.0.0 --port 8002 --proxy-headers --log-config chromadb/log_config.yml --timeout-keep-alive 30` --> results in chroma running on port 80002

Reply:
I can revert this change if that's the preference, so the default `CMD` in `Dockerfile` would be `uvicorn chromadb.app:app --workers 1 --host 0.0.0.0 --port 8000 --proxy-headers --log-config chromadb/log_config.yml --timeout-keep-alive 30` and in `docker_entrypoint.sh` line 10 would be `exec ${@}`

Reply:
@MrZoidberg, thanks for the explanation. I think reconfigurability is important to accommodate more use cases like yours. Security is also important.  Overall, I agree with your reasoning on this. Do you think it merits investigating env vars instead of actual unicorn params? Or perhaps a mix of both.

Here's my reasoning on this:

- Exposing uvicorn params is great for power users who know their stuff, but not so great for users who don't necessarily know a thing about uvicorn, and they shouldn't be forced to learn it to be able to configure a port
- Exposing env vars for common config params, great for less advanced users, limiting for users that want more control over uvicorn

Conclusion - let's meet in the middle - env vars in the `Dockerfile` with sensible defaults. Those same vars can be added to the default `CMD`

wdyt?

Reply:
@tazarov it this something you described?

![image](https://github.com/chroma-core/chroma/assets/126764/3a88800e-4dd9-40ed-b9e3-26eef21a3a1c)

note: expose doesn't do anything and to my knowledge can't be set dynamically other than using build-time ARG.

Reply:
ENV only sets sensible defaults. If the user sets `-e CHROMA_HOST_PORT=8002`, then the ENV will be overwritten by whatever the user sets.

Your suggestion is spot on with what I was thinking.

Reply:
ok will check to make sure it's working with defaults and custom values

Reply:
![image](https://github.com/chroma-core/chroma/assets/126764/089a6ab0-a715-4a98-ac2a-b3d99b8e881c)

works. will push changes soon.

Reply:
done with updates

Reply:
@MrZoidberg, thank you for all the efforts and for sticking with me through this. I think the PR looks excellent and serves your and, I hope, countless other users' use cases without compromising on usability and security and not negatively impacting existing DX.

@beggers have a look, but I think this is good to go after the tests pass.